lib/http.js

```diff
@@ -162,7 +162,9 @@
          * @method splunkjs.Http
          */
         post: function(url, headers, params, timeout, callback) {
-            headers["Content-Type"] = "application/x-www-form-urlencoded";
+            if (!headers.hasOwnProperty("Content-Type")) {
```

Comment:
I think we need to do it a little bit differently, which is going to be a bit of a pain, but we have to in order to handle case sensitivity issues:

1. You should get a list of all the object keys.
2. You should iterate over them, lowercase them, and check if they're equal to "content-type"
3. If any of them are, don't do anything.
4. If none of them are, use the `"application/x-www-form-urlencoded"` value.